code401challenges/src/main/java/stacksAndQueues/Queue.java

```diff
@@ -0,0 +1,55 @@
+package stacksAndQueues;
+
+import java.util.NoSuchElementException;
+
+public class Queue {
+    //instance variables
+    Node front;
+    Node back;
+    Node next;
+
+    //constructor function
+    Queue () {
+        this.front = null;
+        this.next = null;
```

Comment:
Looks like you do not need this.next = null, nor do you need its instance variable above. The .next you use is from the Node class.